### ISSUE TITLE
Pidgin crash on Steam login (ARM)

### ISSUE THREAD

**Issue by PandaBear77** [EionRobb/pidgin-opensteamworks#138]
Hi,
Im using a custom ARM build of opensteamworks, it logins well, but it instantly crash my Pidgin client after login. I do not even see the friends list. My guess is that im using an old build (attached). Could you please give me a updated build for my Raspberry Pi 2 Raspbian? I do not know how to convert the source code to ARM myself.

**Comment by EionRobb** [comment]
The instructions at https://github.com/EionRobb/pidgin-opensteamworks#how-to-build-on-linux should work for you, but I'll see if I can do one too

**Comment by EionRobb** [comment]
Well I tried building it on my own rPi running raspbian but I filled up the SD card with packages.  Let me know how you get on building from source :)

**Comment by PandaBear77** [comment]
pi@raspberrypi:~/pidgin-opensteamworks-1.6.1/steam-mobile $ make && make installPackage glib-2.0 was not found in the pkg-config search path.
Perhaps you should add the directory containing `glib-2.0.pc'
to the PKG_CONFIG_PATH environment variable
No package 'glib-2.0' found
Package json-glib-1.0 was not found in the pkg-config search path.
Perhaps you should add the directory containing `json-glib-1.0.pc'
to the PKG_CONFIG_PATH environment variable
No package 'json-glib-1.0' found
Package purple was not found in the pkg-config search path.
Perhaps you should add the directory containing `purple.pc'
to the PKG_CONFIG_PATH environment variable
No package 'purple' found
Package nss was not found in the pkg-config search path.
Perhaps you should add the directory containing `nss.pc'
to the PKG_CONFIG_PATH environment variable
No package 'nss' found
Package gnome-keyring-1 was not found in the pkg-config search path.
Perhaps you should add the directory containing `gnome-keyring-1.pc'
to the PKG_CONFIG_PATH environment variable
No package 'gnome-keyring-1' found
Package glib-2.0 was not found in the pkg-config search path.
Perhaps you should add the directory containing `glib-2.0.pc'
to the PKG_CONFIG_PATH environment variable
No package 'glib-2.0' found
Package json-glib-1.0 was not found in the pkg-config search path.
Perhaps you should add the directory containing `json-glib-1.0.pc'
to the PKG_CONFIG_PATH environment variable
No package 'json-glib-1.0' found
Package purple was not found in the pkg-config search path.
Perhaps you should add the directory containing `purple.pc'
to the PKG_CONFIG_PATH environment variable
No package 'purple' found
Package nss was not found in the pkg-config search path.
Perhaps you should add the directory containing `nss.pc'
to the PKG_CONFIG_PATH environment variable
No package 'nss' found
gcc -Wall -I. -g -O2 -fPIC -pipe steam_connection.c libsteam.c  -o libsteam.so   -shared
In file included from steam_connection.h:5:0,
                 from steam_connection.c:3:
libsteam.h:9:18: fatal error: glib.h: No such file or directory
 #include <glib.h>
                  ^
compilation terminated.
In file included from libsteam.c:1:0:
libsteam.h:9:18: fatal error: glib.h: No such file or directory
 #include <glib.h>
                  ^
compilation terminated.
Makefile:17: recipe for target 'libsteam.so' failed
make: *** [libsteam.so] Error 1

**Comment by xvitaly** [comment]
> was not found in the pkg-config search path

You must install required development packages first.

**Comment by PandaBear77** [comment]
ANd what are those? Half of the software I was used to install on x86 linux is not available on ARM

**Comment by PandaBear77** [comment]
pi@raspberrypi:~ $ sudo apt-get install json-glib-1.0 glib-2.0 purple gnome-keyring-1 nss
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Note, selecting 'libjson-glib-1.0-common' for regex 'json-glib-1.0'
Note, selecting 'libjson-glib-1.0-0' for regex 'json-glib-1.0'
Note, selecting 'gobject-introspection-glib-2.0' for regex 'glib-2.0'
Note, selecting 'gir1.0-glib-2.0' for regex 'glib-2.0'
Note, selecting 'libqtglib-2.0-0' for regex 'glib-2.0'
Note, selecting 'gir1.2-glib-2.0' for regex 'glib-2.0'
E: Unable to locate package purple
E: Unable to locate package gnome-keyring-1
E: Unable to locate package nss

**Comment by xvitaly** [comment]
> ANd what are those?

I'm Fedora user so I can't answer this question. On Fedora the required packages are json-glib-devel, libpurple-devel, zlib-devel, glib2-devel, libgnome-keyring-devel and nss-devel. Try searching Debian repository for required packages with -dev or -devel suffix.

**Comment by PandaBear77** [comment]
This is resulting in too much hassle for just compiling a program, I abandon. But anyway I thank you both for your time helping me on this matter.
Just for the record:
After hand pick searching the packages you told me, I downloaded them and I got this error:

pi@raspberrypi:~/Desktop/dependences $ sudo dpkg -i *
(Reading database ... 134852 files and directories currently installed.)
Preparing to unpack libcurl4-nss-dev_7.47.0-1_armhf.deb ...
Unpacking libcurl4-nss-dev:armhf (7.47.0-1) over (7.47.0-1) ...
Preparing to unpack libgnome-keyring-dev_3.12.0-1+b1_armhf.deb ...
Unpacking libgnome-keyring-dev (3.12.0-1+b1) over (3.12.0-1+b1) ...
Preparing to unpack libjson-glib-dev_0.14.2-1_armhf.deb ...
Unpacking libjson-glib-dev (0.14.2-1) over (0.14.2-1) ...
Preparing to unpack libpackagekit-glib2-14_0.7.6-3_armhf.deb ...
Unpacking libpackagekit-glib2-14:armhf (0.7.6-3) over (0.7.6-3) ...
Preparing to unpack libpurple-dev_2.10.10-1~deb7u1_all.deb ...
Unpacking libpurple-dev (2.10.10-1~deb7u1) over (2.10.10-1~deb7u1) ...
Preparing to unpack zlib1g-dev_1.2.7.dfsg-13_armhf.deb ...
Unpacking zlib1g-dev:armhf (1:1.2.7.dfsg-13) over (1:1.2.7.dfsg-13) ...
dpkg: dependency problems prevent configuration of libcurl4-nss-dev:armhf:
 libcurl4-nss-dev:armhf depends on libcurl3-nss (= 7.47.0-1); however:
  Package libcurl3-nss is not installed.

dpkg: error processing package libcurl4-nss-dev:armhf (--install):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of libgnome-keyring-dev:
 libgnome-keyring-dev depends on gir1.2-gnomekeyring-1.0 (= 3.12.0-1+b1); however:
  Package gir1.2-gnomekeyring-1.0 is not installed.
 libgnome-keyring-dev depends on libglib2.0-dev (>= 2.16); however:
  Package libglib2.0-dev is not installed.

dpkg: error processing package libgnome-keyring-dev (--install):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of libjson-glib-dev:
 libjson-glib-dev depends on libjson-glib-1.0-0 (= 0.14.2-1); however:
  Version of libjson-glib-1.0-0:armhf on system is 1.0.2-1.
 libjson-glib-dev depends on libglib2.0-dev; however:
  Package libglib2.0-dev is not installed.
 libjson-glib-dev depends on gir1.2-json-1.0 (= 0.14.2-1); however:
  Package gir1.2-json-1.0 is not installed.

dpkg: error processing package libjson-glib-dev (--install):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of libpackagekit-glib2-14:armhf:
 libpackagekit-glib2-14:armhf depends on libarchive12; however:
  Package libarchive12 is not installed.
 packagekit (1.0.1-2) breaks libpackagekit-glib2-14 (<= 0.7.6-4) and is installed.
  Version of libpackagekit-glib2-14:armhf to be configured is 0.7.6-3.

dpkg: error processing package libpackagekit-glib2-14:armhf (--install):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of libpurple-dev:
 libpurple-dev depends on libpurple0 (<< 2.10.10+1~); however:
  Version of libpurple0 on system is 2.10.11-1.
 libpurple-dev depends on libglib2.0-dev; however:
  Package libglib2.0-dev is not installed.
 libpurple-dev depends on libdbus-glib-1-dev; however:
  Package libdbus-glib-1-dev is not installed.

dpkg: error processing package libpurple-dev (--install):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of zlib1g-dev:armhf:
 zlib1g-dev:armhf depends on zlib1g (= 1:1.2.7.dfsg-13); however:
  Version of zlib1g:armhf on system is 1:1.2.8.dfsg-2.

dpkg: error processing package zlib1g-dev:armhf (--install):
 dependency problems - leaving unconfigured
Processing triggers for man-db (2.7.0.2-5) ...
Errors were encountered while processing:
 libcurl4-nss-dev:armhf
 libgnome-keyring-dev
 libjson-glib-dev
 libpackagekit-glib2-14:armhf
 libpurple-dev
 zlib1g-dev:armhf
